src/bin/upgrade/main.rs

```diff
@@ -114,6 +114,10 @@ struct Args {
     #[clap(long, conflicts_with = "to-lockfile")]
     skip_compatible: bool,
 
+    /// Do not increase dependency requirement precision.
+    #[clap(long)]
+    preserve_precision: bool,
```

Comment:
Let's give this a try of making this not behind a flag.  We can always start with it being unstable and use #614 to opt-in.